api/trace.proto

```diff
@@ -52,3 +53,12 @@ message ZipkinConfig {
   // /api/v1/spans, which is the default value.
   string collector_endpoint = 2 [(validate.rules).string.min_bytes = 1];
 }
+
+message DynamicConfig {
```

Comment:
Can we call this `DynamicOtConfig`? I think it's unlikely but it's possible we would eventually support loadable modules of some other type. Same for above perhaps `envoy.dynamic.ot` for the type?